File: week_2/pset_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int calculate_index(string text);

int main(void)
{
    int index = 0;

    string text = get_string("Text: ");
    index = calculate_index(text);

    if (index < 1) {
        printf("Before Grade 1\n");
    }
    else if (index > 16) {
        printf("Grade 16+\n");
    }
    else {
        printf("Grade %i\n", index);
    }

}

int calculate_index(string text) {
    float index = 0;
    float letters = 0;
    float words = 1;
    float sentenses = 0;
    float l = 0;
    float s = 0;
    int result = 0;

    for (int i = 0, lenght = strlen(text); i < lenght; i++) {
        if (isupper(text[i])) {
            text[i] = tolower(text[i]);
        }

        if (text[i] == 32) {
            words++;
        }
        else if (text[i] == 46 || text[i] == 63 || text[i] == 33) {
            sentenses++;
        }
        else if (text[i] >= 97 && text[i] <= 122) {
            letters++;
        }
    }

    l = (letters / words) * 100.0;
    s = (sentenses / words) * 100.0;

    index = 0.0588 * l - 0.296 * s - 15.8;

    return round(index);
}

File: week_6/pset_6/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    text = text.lower()
    index = calculateIndex(text)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

def calculateIndex(text):
    words = (0 + 1)
    sentenses = 0
    letters = 0

    for i in range(len(text)):
        if text[i] == " ":
            words += 1
        elif text[i] == "." or text[i] == "!" or text[i] == "?":
            sentenses += 1
        elif text[i] >= "a" and text[i] <= "z":
            letters += 1

    l = (letters / words) * 100
    s = (sentenses / words) * 100

    index = 0.0588 * l - 0.296 * s - 15.8

    return round(index)

main()